enhancements/machine-api/cluster-autoscaler-operator.md

```diff
@@ -0,0 +1,290 @@
+---
+title: cluster-autoscaler-operator
+authors:
+  - "@JoelSpeed"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-05-19
+last-updated: 2020-05-19
+status: implemented
+see-also:
+replaces:
+superseded-by:
+---
+
+# Cluster Autoscaler Operator
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Provide a declarative way for users to enable autoscaling of compute resources by providing the Cluster Autoscaler
+as a Deployment with opinionated default options.
+
+Provide a way for users to opt individual MachineSets into autoscaling and to set boundaries for the size of the
+MachineSets that they wish to autoscale.
+
+## Motivation
+
+### Goals
+
+- Enable users to create Cluster Autoscaler with opinionated defaults
+- Enable users to autoscale MachineSets
+- Enable users to set a maximum size of their cluster
+- Enable users to set maximum and minimum sizes for each of their MachineSets
+
+### Non-Goals
+
+- Integrate the Cluster Autoscaler with Machine API (This will be a separate proposal)
+- Enable autoscaling by default for new OpenShift clusters
+- Provide options for autoscaling across multiple MachineSets
+
+## Proposal
+
+A new controller will be developed to manage the lifecycle of the Cluster Autoscaler application and to manage
+assigning size limits to MachineSets to enable the Cluster Autoscaler to autoscale desired MachineSets.
```

Comment:
This is super-sparse.  Can you fill in more information about expected user workflows or something?  The later _User Stories_ are good for motivation, and _Implementation Details_ is good for devs, but this section should have some sort of sketch of what admin-facing openshift-docs content will look like.

Reply:
Good suggestion, I'll look at expanding this